src/proj.rs

```diff
@@ -1064,6 +1065,64 @@ impl Proj {
             Err(ProjError::Projection(error_message(err)?))
         }
     }
+
+    /// Return the projjson representation of a transformation
+    ///
+    /// # Safety
+    /// This method contains unsafe code.
+    pub fn to_projjson(
+        &self,
+        multiline: Option<bool>,
+        indentation_width: Option<usize>,
+        schema: Option<&str>,
+    ) -> Result<String, ProjError> {
+        let opts_p = match (multiline, indentation_width, schema) {
+            (None, None, None) => None,
+            _ => {
+                let mut opts = vec![];
+                if let Some(multiline) = multiline {
+                    if multiline {
+                        opts.push(CString::new("MULTILINE=YES")?);
+                    } else {
+                        opts.push(CString::new("MULTILINE=NO")?);
+                    }
+                }
+                if let Some(indentation_width) = indentation_width {
+                    opts.push(CString::new(format!(
+                        "INDENTATION_WIDTH={}",
+                        indentation_width
+                    ))?);
+                }
+                if let Some(schema) = schema {
+                    opts.push(CString::new(format!("SCHEMA={}", schema))?);
+                }
+
+                Some(opts)
+            }
+        };
+
+        unsafe {
+            // NOTE: we can't create the pointer to the vec of strings too early! As the [docstring
+            // of `as_ptr`
+            // mentions](https://doc.rust-lang.org/std/ffi/struct.CStr.html#method.as_ptr):
+            // > It is your responsibility to make sure that the underlying memory is not freed too
+            // > early. For example, the following code will cause undefined behavior when ptr is
+            // > used inside the unsafe block:
+            let out_ptr = if let Some(opts_p) = opts_p {
+                let opts_c = opts_p.iter().map(|x| x.as_ptr()).collect::<Vec<_>>();
+                proj_as_projjson(self.ctx, self.c_proj, opts_c.as_ptr())
```

Comment:
This probably needs to finish with a null.

Reply:
Sorry what does `finish with a null` mean?

Reply:
Push a ptr::null at the end, otherwise it's unterminated.

Reply:
I'm confused too. At the end of what?

Reply:
At the end of opts_c.